### ISSUE TITLE
[bug]: UI should be able to display any supported image types in marketplace

### ISSUE THREAD

**Issue by elvece** [Start9Labs/embassy-os#1318]
Currently, the UI is hardcoded to always display the base64 encoded images as pngs: `data:image/png;base64`

This is incorrect, as developers can use the following icon types: `svg | jpg | jpeg | png | gif`

Solution:
Decode the base64 response from the marketplace response to determine the file type. Use this file type to populate the data URL.

**Comment by ProofOfKeags** [comment]
Do we encode file type in the base64 data?

**Comment by elvece** [comment]
> Do we encode file type in the base64 data?

I am unable to find where this explicitly happens in code, including looking at the source for [encodeBase64](https://hackage.haskell.org/package/base64-0.4.2.3/docs/src/Data.ByteString.Base64.html#encodeBase64). However, if I copy the generated base64 string from an icon in the `/package/v0/index` response, I am able to determine its mime type via the "hacky" way listed above. 

Related: https://github.com/Start9Labs/registry/issues/99

**Comment by ProofOfKeags** [comment]
> I am able to determine its mime type via the "hacky" way listed above.

yeah most filetypes have headers that make it easy to tell what the format is. But yeah we can change over to a data URL scheme.

**Comment by elvece** [comment]
Welp @dr-bonez I completely missed this one as being part of this milestone since it was still assigned to Drew. It is not a blocking concern, and ideally should be resolved with a small change in the registry that will make the FE implementation much reliable. Punting to next release.

**Comment by elvece** [comment]
@ProofOfKeags needs to be implemented for 0.3.1 release to support this feature https://github.com/Start9Labs/registry/issues/99
